File: Assets/SayKit/Internal/Plugins/iOS/SayKit/SayKitVersionManager.m
//
//  SayKitVersionManager.m
//  Unity-iPhone
//
//  Created by Timur Dularidze on 9/4/19.
//

#import "SayKitVersionManager.h"

@implementation SayKitVersionManager


// Manager to be used for methods that do not require a specific adunit to operate on.
+ (SayKitVersionManager*)sharedInstance
{
    static SayKitVersionManager* sharedSayVersionManager = nil;
    
    if (!sharedSayVersionManager)
    {
        sharedSayVersionManager = [[SayKitVersionManager alloc] init];
    }
    
    return sharedSayVersionManager;
}

static NSMutableDictionary * sdkVersionDictionary;


- (void) trackAdNetworkVersions {
    
    @try {
        
        NSString* sayPromoVersion =  [[SPAdManager sharedInstance] getVersion];
        [self trackSDKVersion:@"sdk_saypromo" version:sayPromoVersion];
        
        NSString* adColonyVersion = [AdColony getSDKVersion];
        [self trackSDKVersion:@"sdk_adColony" version:adColonyVersion];
        
        NSString* appLovinVersion = ALSdk.version;
        [self trackSDKVersion:@"sdk_applovin" version:appLovinVersion];
        
        NSString* facebookVersion = FB_AD_SDK_VERSION;
        [self trackSDKVersion:@"sdk_facebook" version:facebookVersion];
        
        NSString* ironSourceVersion = [IronSource sdkVersion];
        [self trackSDKVersion:@"sdk_ironsource" version:ironSourceVersion];
        
        NSString* byteDanceVersion = [BUAdSDKManager SDKVersion];
        [self trackSDKVersion:@"sdk_bytedance" version:byteDanceVersion];
        
        NSString* unityAdsVersion = [UnityAds getVersion];
        [self trackSDKVersion:@"sdk_unity" version:unityAdsVersion];
        
        NSString* vungleVersion =VungleSDKVersion;
        [self trackSDKVersion:@"sdk_vungle" version:vungleVersion];
        
        NSString* mintegralVersion = MTGSDK.sdkVersion;
        [self trackSDKVersion:@"sdk_mintegral" version:mintegralVersion];
        
        NSString* fyberVersion = IASDKCore.sharedInstance.version;
        [self trackSDKVersion:@"sdk_fyber" version:fyberVersion];
        
        NSString* inMobiVersion = [IMSdk getVersion];
        [self trackSDKVersion:@"sdk_inmobi" version:inMobiVersion];
        
        
        [self trackSDKVersion:@"sdk_yandex" version:@"4.3.0"];
        
        NSString* mytargerVersion = [MTRGVersion currentVersion];
        [self trackSDKVersion:@"sdk_mytarget" version:fyberVersion];
        
        
//        NSString* apsVersion = DTBAds.version;
//        NSArray * apsVersionArray = [[NSArray alloc] init];
//        apsVersionArray = [apsVersion componentsSeparatedByString:@"-"];
//        apsVersion = apsVersionArray[apsVersionArray.count - 1];
        
        [self trackSDKVersion:@"sdk_aps" version:@"3.4.4"];//apsVersion
        
        NSString* admobVersion =  [NSString stringWithUTF8String:(char*)GoogleMobileAdsVersionString];
        NSArray * admobVersionArray = [[NSArray alloc] init];
        admobVersionArray = [admobVersion componentsSeparatedByString:@"-v"];
        admobVersion = admobVersionArray[admobVersionArray.count - 1];
        
        [self trackSDKVersion:@"sdk_admob" version:admobVersion];
        
    } @catch (NSException *exception) {
        [[SayKitLog sharedInstance] Log:[NSString stringWithFormat:@"checkAdNetworkVersions error: %@\n", exception.reason]];
    }
    
}

- (void) trackSDKVersion:(NSString*)sdkName
                 version:(NSString*)version
{
    [SayKitEvent.sharedInstance track:sdkName
                               param1:0
                               param2:0
                                extra:version
                               param3:0
                               param4:0
                               extra2:@""
                                  tag:@""];
}


@end

File: Assets/SayKit/Internal/Plugins/Android/src/SayKitVersionManager.java
package by.saygames;

import com.amazon.device.ads.AdRegistration;

import com.bytedance.sdk.openadsdk.TTAdSdk;
import com.fyber.inneractive.sdk.external.InneractiveAdManager;
import com.inmobi.sdk.InMobiSdk;
import com.ironsource.mediationsdk.utils.IronSourceUtils;
import com.mbridge.msdk.out.MBConfiguration;

import com.my.target.common.MyTargetVersion;
import com.ogury.sdk.Ogury;
import com.saypromo.SPAdManager;
import com.unity3d.ads.UnityAds;
import com.applovin.sdk.AppLovinSdk;

public class SayKitVersionManager {

    public static void trackSDKVersions() {
        try {

            //TODO Update sdk_adcolony
            SayKitEvents.track("sdk_adcolony", 0, 0, "4.6.3");
            SayKitEvents.track("sdk_applovin", 0, 0, AppLovinSdk.VERSION);
            //TODO Update sdk_admob
            SayKitEvents.track("sdk_admob", 0, 0, "20.3.0");
            SayKitEvents.track("sdk_facebook", 0, 0, com.facebook.ads.BuildConfig.VERSION_NAME);
            SayKitEvents.track("sdk_ironsource", 0, 0, IronSourceUtils.getSDKVersion());
            SayKitEvents.track("sdk_bytedance", 0, 0, TTAdSdk.getAdManager().getSDKVersion());
            SayKitEvents.track("sdk_unity", 0, 0, UnityAds.getVersion());
            SayKitEvents.track("sdk_vungle", 0, 0, com.vungle.warren.BuildConfig.VERSION_NAME);
            SayKitEvents.track("sdk_saypromo", 0, 0, SPAdManager.getSDKVersion());
            SayKitEvents.track("sdk_fyber", 0, 0, InneractiveAdManager.getVersion());
            SayKitEvents.track("sdk_inmobi", 0, 0, InMobiSdk.getVersion());

            //TODO Update sdk_yandex
            SayKitEvents.track("sdk_yandex", 0, 0, "4.1.0");
            SayKitEvents.track("sdk_mytarget", 0, 0, MyTargetVersion.VERSION);
            SayKitEvents.track("sdk_ogury", 0, 0, Ogury.getSdkVersion());


            String apsVersion = AdRegistration.getVersion();
            String[] apsVersionArray = apsVersion.split("-");
            apsVersion = apsVersionArray[apsVersionArray.length - 1];

            SayKitEvents.track("sdk_aps", 0, 0, apsVersion);

            String mintegralVersion = MBConfiguration.SDK_VERSION;
            String[] mintegralVersionArray = mintegralVersion.split("_");
            mintegralVersion = mintegralVersionArray[mintegralVersionArray.length - 1];

            SayKitEvents.track("sdk_mintegral", 0, 0, mintegralVersion);

        } catch (Exception e) {
            SayKitLog.Log("i", "[SayKitVersionManager]",
                    "Error getting versions ", e);
        }
    }

}
